packages/backend/src/rpc/dispatcher.ts

```diff
@@ -66,17 +67,18 @@ export class RPCDispatcher extends BaseService {
   }
 
   private fail = (e: Error) => {
+    let handler = this.getSingleton(RPCErrorHandlers).handle(e);
+    if (handler) {
+      console.error(`RPC call failed for "${this.rpcPath}":`);
+      return this.jsonFail(handler.code, handler.message, handler.data);
+    }
```

Comment:
Tests are missing. We could create a mocked error, and handler for it (more than one error/handler pairs), then write a test that proves this works.

Reply:
@gphfour lets see if this needs expanding
https://github.com/hfour/h4bff/blob/2f859e64e4e4eb9d07e106d4466de00e887c3229/packages/backend/src/rpc/dispatcher.spec.ts#L285